### PR TITLE
Feature/comments

### PR DESCRIPTION
**Pull request by martinagrzybowska**
General:
- adding infinite scroll to Feed, Explore and UserProfile posts
- fixing height of follows and following pages (was stuck at h-screen)
- fixing component not reloading when switching route from one user to another

Feed:
- transforming Events store to Feed store
- adding Tree structure to Feed store, in order to be able to display posts in tree-like structure of replies
- adding recursive adding/parsing of threads of events

Posts:
- responsibility for checking whether event is liked or not was moved here
- reaction buttons now show action name on hover
- adding recursive component PostThread

NostrHandler:
- adding reply functionality

### CODE REVIEW

File: src/frontend/src/routes/(app)/post/[id]/+page.svelte [hunk 1]
```diff
@@ -30,3 +35,90 @@
...
+    <div class="flex justify-end">
+      <button class="btn variant-filled mt-2">Post reply</button>
+    </div>
+  </form>
+</div>
+<div class="m-5">
+  <!-- Only display comment to the post/event displayed, to see comments on the comment user needs to click on the comment to go down the rabbit hole -->
+  <div class="grid grid-cols-[auto_1fr] gap-2 my-4">
+    <!-- Link to the author -->
+    <a href="#">
```
Comment (viktorkovarik) [inline src/frontend/src/routes/(app)/post/[id]/+page.svelte:57]:
```suggestion
    <a href="#TODO">
```
+ other places, it breaks stuff then

**Comment by viktorkovarik** [comment]
We should use white loading animations for dark mode.
Screenshot 2024-01-28 at 20 54 07
Screenshot 2024-01-28 at 20 54 26

File: src/frontend/src/lib/components/post/Post.svelte [hunk 2]
```diff
@@ -45,88 +51,147 @@
...
-      event &&
-      $authUser.authState !== AuthStates.ANONYMOUS
-    ) {
+    if ($authUser.nostr && event && isAuthenticated) {
       if (isLiked) {
         await $authUser.nostr.dislikeEvent(event);
+        isLiked = false;
       } else {
         await $authUser.nostr.likeEvent(event);
+        isLiked = true;
```
Comment (viktorkovarik) [inline src/frontend/src/lib/components/post/Post.svelte:60]: add some loading when clicked on like button because liking can take a while

Comment (viktorkovarik) [reply src/frontend/src/lib/components/post/Post.svelte:60]: this was resolved by latest commit, thanks for fixing it 👏 🎉

File: src/frontend/src/routes/(app)/post/[id]/+page.svelte (hunk 1 above)
Comment (viktorkovarik) [reply src/frontend/src/routes/(app)/post/[id]/+page.svelte]: this was also fixed by latest commit

**Comment by lukasvozda** [comment]
@martinagrzybowska  is it a complex change to see commented post and subsequent comments of the comment we open the post detail for? Right now when I click on the comment post, I only see the 1 comment.
image

**Comment by martinagrzybowska** [comment]
> @martinagrzybowska is it a complex change to see commented post and subsequent comments of the comment we open the post detail for? Right now when I click on the comment post, I only see the 1 comment. image

should be fixed

**Review by viktorkovarik — APPROVED** [review]

**Pull request merged**